src/codegen/doc/mod.rs

```diff
@@ -330,10 +330,9 @@ fn create_object_doc(w: &mut dyn Write, env: &Env, info: &analysis::object::Info
 
     if has_trait {
         write_item_doc(w, &ty_ext, |w| {
-            writeln!(w, "Trait containing all `{}` methods.", ty.name)?;
+            writeln!(w, "Trait containing all [`struct@{}`] methods.", ty.name)?;
 
-            let mut implementors = Some(info.type_id)
-                .into_iter()
+            let mut implementors = std::iter::once(info.type_id)
```

Comment:
Just a cleanup :)... But a more pressing question: do we want to repeat the type name right here, as it is already implied by the `Trait containing all {myself} methods` line?